### DISCUSSION TITLE
How to get the position relative to the control in a touch event

### DISCUSSION THREAD

**Discussion by pauldendulk** [unoplatform/uno#15421] [Q&A] [answered]
Hi there, I am working on Pinch for Mapsui.Uno.WinUI and I can't figure out the way to calculate the position of the touch event relative to the MapControl. 

This is the event:

`private void OnManipulationDelta(object sender, ManipulationDeltaRoutedEventArgs e)`

e.Position is relative to the application not to the MapControl.

My code is here:
https://github.com/Mapsui/Mapsui/pull/2484/files#diff-1f65ba136643bcb7ba7718e26a209f067842cb5c3cfcc8aace4fa97fed13757eR242

**Comment by dr1rrb** [comment, marked as answer]
Hey @pauldendulk 👋

You can convert from absolute coordinates to a relative coordinates system using the [`TransformToVisual`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.transformtovisual?view=winrt-22621). In your case you can do something like :

```csharp
var relPosition = this.TransformToVisual(null).Inverse.TransformPoint(e.Position);
```

Also, I saw that you are accumulating values by your own (e.g. `_totalPinchRotation + e.Delta.Rotation`), you could consider to use the [`e.Cummulative.Rotation`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.input.manipulationdeltaroutedeventargs.cumulative?view=winrt-22621#windows-ui-xaml-input-manipulationdeltaroutedeventargs-cumulative) instead.

BTW I'm a bit surprised that the `Position` is expressed in absolute coordinates, but the documentation specifies only ["The point from which the manipulation originated."](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.input.manipulationdeltaroutedeventargs.position?view=winrt-22621#windows-ui-xaml-input-manipulationdeltaroutedeventargs-position). In order to exclude any bug here, you should validate that it's the case on all platform you are targeting.

**Reply by pauldendulk** [reply]
Thanks for your quick replay. Copy-pasting your code solved my problem right away!

In a later commit I move the _totalPinchRotation tracking inside the Navigator. I could have used the cumulative rotation, but the other platforms don't have that. For me the big puzzle is to make it work in a similar way on all platforms.

I don't see an option to test on other platforms. I don't have a Mac and no touch screen on my laptops. I tested on Android deploying from Windows. Also I suspect it is not a bug, the documentation says it is the point from which the manipulation originated, but that point is just relative to the full screen, not the component.

**Reply by sasakrsmanovic** [reply]
@pauldendulk Is using Mac in the cloud an option for you?

**Reply by pauldendulk** [reply]
Might be, so far I never tried it. Could I deploy from a Mac in the cloud to my hardware iPad? When I had a Mac (it broke two years ago) I used it only now an then and my biggest problem was to keep it up to date with all the latest software and tooling. I guess that is also a problem with a Mac in the cloud.

**Reply by sasakrsmanovic** [reply]
A few from our team are / were using it - @MartinZikmund @Youssef1313  @ramezgerges . They may be able to provide some insights for your use case.

**Reply by ramezgerges** [reply]
I haven't tried deploying to a hardware iPad as I don't have one, but you should be able to connect a local VS Code (or VS but with less stability) instance to a remote Mac and then test on an emulator and/or a physical device (I tried this for an emulated iPhone, not an iPad, but it should be similar). If that fails, you can try to run VS/VSCode on the remote machine directly and test/debug there. 

Since the remote Mac is just a rented instance, you will indeed have to maintain your own software updates.

Our own @nickrandolph wrote an awesome article on this:
https://nicksnettravels.builttoroam.com/unoplatform-vscode-extensions/
